cleanlab_pinecone_rag_workflow/Cleanlab_Pinecone_RAG.ipynb

```diff
@@ -0,0 +1,5263 @@
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>index</th>\n",
+       "      <th>filename</th>\n",
+       "      <th>text</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>0</td>\n",
+       "      <td>documents/Blackstone-Third-Quarter-2023-Invest...</td>\n",
+       "      <td>Blackstone Third Quarter 2023 Investor Call Oc...</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>1</td>\n",
+       "      <td>documents/8k-nike.pdf</td>\n",
+       "      <td>SECURITIES AND EXCHANGE COMMISSIONFORM 8-K Cur...</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>2</td>\n",
+       "      <td>documents/FY24-Q1-NIKE-Press-Release.pdf</td>\n",
+       "      <td>NaN</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>3</td>\n",
+       "      <td>documents/10-K 2022-Apple2.pdf</td>\n",
+       "      <td>The future principal payments for the Company’...</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>4</td>\n",
+       "      <td>documents/q3-fy22-earnings-presentation.pdf</td>\n",
+       "      <td>Financial\\tpresentation\\tto\\t accompany\\tmanag...</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   index                                           filename  \\\n",
+       "0      0  documents/Blackstone-Third-Quarter-2023-Invest...   \n",
+       "1      1                              documents/8k-nike.pdf   \n",
+       "2      2           documents/FY24-Q1-NIKE-Press-Release.pdf   \n",
+       "3      3                     documents/10-K 2022-Apple2.pdf   \n",
+       "4      4        documents/q3-fy22-earnings-presentation.pdf   \n",
+       "\n",
+       "                                                text  \n",
+       "0  Blackstone Third Quarter 2023 Investor Call Oc...  \n",
+       "1  SECURITIES AND EXCHANGE COMMISSIONFORM 8-K Cur...  \n",
+       "2                                                NaN  \n",
+       "3  The future principal payments for the Company’...  \n",
+       "4  Financial\\tpresentation\\tto\\t accompany\\tmanag...  "
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "69f026d6-fc2d-4b9a-ac5d-eb227ec9462b",
+   "metadata": {},
+   "source": [
+    "## Analyze Documents Data\n",
```

Comment:
what is the point of this whole length-analysis section?  It seems unnecesary, so I recommend cutting it